Podcasts app crashes -> NullReferenceException => solution Disable XAML Hot Reload in Tools->Options

I watched the [.NET Podcasts app - Local Development Walkthrough](https://www.youtube.com/watch?v=FJowBj-DBgI) video on Youtube. When I tried to run the .NET Maui app on my mobile phone, the app threw a NullReferenceException.

After checking the Device Log I found it had something to do with XAML Hot Reload.  After disabling XAML Hot Reload in Tools->Options in Visual Studio 2022 Preview, I could run the Podcasts app.

DEVICE LOG
01-07 22:55:41.354	pixel_2_pie_9_0_-_api_28	Error	6645	mono-rt	[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
   at Xamarin.HotReload.Xaml.VisualTreeService.DiagnosticVisualTreeService.Dispose(Boolean isDisposing) in D:\a\_work\1\s\HotReload\Source\Xamarin.HotReload.Xaml\VisualTreeService\DiagnosticVisualTreeService.cs:line 75
   at Xamarin.HotReload.Xaml.VisualTreeService.VisualTreeService.Finalize() in D:\a\_work\1\s\HotReload\Source\Xamarin.HotReload.Xaml\VisualTreeService\VisualTreeService.cs:line 138

--- a/Directory.build.props
+++ b/Directory.build.props
@@ -1,5 +1,5 @@
 <Project>
   <PropertyGroup>
-	  <MauiV>6.0.101-preview.11.2082</MauiV>
+	  <MauiV>6.0.101-preview.11.2349</MauiV>
   </PropertyGroup>
 </Project>

--- a/src/Mobile/Microsoft.NetConf2021.Maui.csproj
+++ b/src/Mobile/Microsoft.NetConf2021.Maui.csproj
@@ -47,8 +47,8 @@
 
   <ItemGroup Condition="$(TargetFramework.Contains('-windows'))">
     <!-- Required - WinUI does not yet have buildTransitive for everything -->
-    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0-preview3" />
-    <PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.0.29-preview3" />
+    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0" />
+    <PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.0.30" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/Mobile/Platforms/Android/Services/MediaPlayerService.cs
+++ b/src/Mobile/Platforms/Android/Services/MediaPlayerService.cs
@@ -11,6 +11,7 @@ using Android.Graphics;
 using System;
 using System.Threading.Tasks;
 using System.Collections.Generic;
+using Microsoft.Maui.Platform;
 
 namespace Microsoft.NetConf2021.Maui.Platforms.Android.Services
 {

--- a/src/Mobile/Services/EssentialsExtensions.cs
+++ b/src/Mobile/Services/EssentialsExtensions.cs
@@ -18,8 +18,6 @@ namespace Microsoft.Maui.Essentials
         IEssentialsBuilder OnAppAction(Action<AppAction> action);
 
         IEssentialsBuilder UseVersionTracking();
-
-        IEssentialsBuilder UseLegacySecureStorage();
     }
 
     public static class EssentialsExtensions
@@ -121,8 +119,6 @@ namespace Microsoft.Maui.Essentials
 
 #if WINDOWS
 				Platform.MapServiceToken = _essentialsBuilder.MapServiceToken;
-#elif __ANDROID__
-				SecureStorage.LegacyKeyHashFallback = _essentialsBuilder.UseLegaceSecureStorage;
 #endif
 
                 AppActions.OnAppAction += HandleOnAppAction;
@@ -155,7 +151,6 @@ namespace Microsoft.Maui.Essentials
             internal bool TrackVersions;
 
 #pragma warning disable CS0414 // Remove unread private members
-            internal bool UseLegaceSecureStorage;
             internal string MapServiceToken;
 #pragma warning restore CS0414 // Remove unread private members
 
@@ -180,12 +175,6 @@ namespace Microsoft.Maui.Essentials
             public IEssentialsBuilder UseVersionTracking()
             {
                 TrackVersions = true;
-                return this;
-            }
-
-            public IEssentialsBuilder UseLegacySecureStorage()
-            {
-                UseLegaceSecureStorage = true;
                 return this;
             }
         }

--- a/src/MobileBlazor/mauiapp/NetPodsMauiBlazor.csproj
+++ b/src/MobileBlazor/mauiapp/NetPodsMauiBlazor.csproj
@@ -43,8 +43,8 @@
 
 	<ItemGroup Condition="$(TargetFramework.Contains('-windows'))">
 		<!-- Required - WinUI does not yet have buildTransitive for everything -->
-		<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0-preview3" />
-		<PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.0.29-preview3" />
+		<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0" />
+		<PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.0.30" />
 	</ItemGroup>
 
 	<PropertyGroup Condition="$(TargetFramework.Contains('-windows'))">

--- a/src/Services/ListenTogether/ListenTogether.Application/ListenTogether.Application.csproj
+++ b/src/Services/ListenTogether/ListenTogether.Application/ListenTogether.Application.csproj
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.1" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="6.0.0" />
   </ItemGroup>
 

--- a/src/Services/ListenTogether/ListenTogether.Hub/ListenTogether.Hub.csproj
+++ b/src/Services/ListenTogether/ListenTogether.Hub/ListenTogether.Hub.csproj
@@ -10,12 +10,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.0">
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.1">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.1" />
     <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.14.0" />
   </ItemGroup>
 

--- a/src/Services/ListenTogether/ListenTogether.Infrastructure/ListenTogether.Infrastructure.csproj
+++ b/src/Services/ListenTogether/ListenTogether.Infrastructure/ListenTogether.Infrastructure.csproj
@@ -7,9 +7,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.1" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="6.0.0" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="6.0.0" />
     <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />

--- a/src/Services/Podcasts/Podcast.API/Podcast.API.csproj
+++ b/src/Services/Podcasts/Podcast.API/Podcast.API.csproj
@@ -11,11 +11,11 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Storage.Queues" Version="12.8.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.0">
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.1">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.1" />
     <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.14.0" />
     <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3" />
   </ItemGroup>

--- a/src/Services/Podcasts/Podcast.Infrastructure/Podcast.Infrastructure.csproj
+++ b/src/Services/Podcasts/Podcast.Infrastructure/Podcast.Infrastructure.csproj
@@ -7,12 +7,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.0">
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.1">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-	<PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
+	<PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.1" />
   </ItemGroup>
 
 </Project>

--- a/src/Services/Podcasts/Podcast.Ingestion.Worker/Podcast.Ingestion.Worker.csproj
+++ b/src/Services/Podcasts/Podcast.Ingestion.Worker/Podcast.Ingestion.Worker.csproj
@@ -11,8 +11,8 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Storage.Queues" Version="12.8.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.1" />
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.0" />
     <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
     <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.14.0" />

--- a/src/Services/Podcasts/Podcast.Updater.Worker/Podcast.Updater.Worker.csproj
+++ b/src/Services/Podcasts/Podcast.Updater.Worker/Podcast.Updater.Worker.csproj
@@ -14,8 +14,8 @@
     <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
     <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.14.0" />
     <PackageReference Include="System.ServiceModel.Syndication" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.1" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/Web/Client/Podcast.Client.csproj
+++ b/src/Web/Client/Podcast.Client.csproj
@@ -9,8 +9,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.0" />
-    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="6.0.0" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.1" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="6.0.1" PrivateAssets="all" />
     <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
   </ItemGroup>
 

--- a/src/Web/Components/Podcast.Components.Maui.csproj
+++ b/src/Web/Components/Podcast.Components.Maui.csproj
@@ -20,8 +20,8 @@
 	</ItemGroup>
 
 	<ItemGroup>
-		<PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.0" />
-		<PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="6.0.0" />
+		<PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.1" />
+		<PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="6.0.1" />
 	</ItemGroup>
 
 </Project>

--- a/src/Web/Components/Podcast.Components.csproj
+++ b/src/Web/Components/Podcast.Components.csproj
@@ -10,8 +10,8 @@
 	</ItemGroup>
 
 	<ItemGroup>
-		<PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.0" />
-		<PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="6.0.0" />
+		<PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.1" />
+		<PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="6.0.1" />
 	</ItemGroup>
 
 </Project>

--- a/src/Web/Pages/Podcast.Pages.Maui.csproj
+++ b/src/Web/Pages/Podcast.Pages.Maui.csproj
@@ -20,7 +20,7 @@
 	</ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.1" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/Web/Pages/Podcast.Pages.csproj
+++ b/src/Web/Pages/Podcast.Pages.csproj
@@ -11,7 +11,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="6.0.1" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/Web/Server/Podcast.Server.csproj
+++ b/src/Web/Server/Podcast.Server.csproj
@@ -10,7 +10,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Server" Version="6.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Server" Version="6.0.1" />
     <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.14.0" />
   </ItemGroup>
 

